#1: implement layer dependency violations

vlad961 opened the pull request:
DO NOT MERGE.
This fork branch contains classes demonstrating archunit layer dependency rules violations.

#1 
#8
#13

vlad961 committed:
Violation of L9-Logic layer doesnt dep. on service

vlad961 committed:
Violation of L1 - Common Layer doesnt depend
on any other layer.

vlad961 committed:
Added sample client.
Removed rules from ArchitectureTest.java.
Because they were too restrictive.
Violation of L2 - Client layer doesnt depend
on any other layer.

vlad961 committed:
Violationf of L3.
Client Layer doesnt depend on logic layer.

vlad961 committed:
Violation of L4.
Client Layer doesnt depend on dataaccess layer.

vlad961 committed:
Refactored L9Violation class name for consistency.
Added Violation of L8.
Service layer doesnt depend on dataaccess layer.

vlad961 committed:
Violation of L10.
Dataaccess layer doesnt depend on service layer.

vlad961 committed:
Refactored L10: Added comment.
Added Violation of L12.
Dataaccess layer doesnt depend on logic layer.

vlad961 committed:
Added sample import script.
Added sample tasklist and taskitem dump file.

vlad961 committed:
Violation of L5.
Client layer doesnt depend on batch layer.

vlad961 committed:
Violation of L6.
Service layer doesnt depend on batch layer.

vlad961 committed:
Violation of L7.
Batch layer doesnt depend on service layer.

vlad961 committed:
Violation of L11.
Batch layer doesnt depend on dataaccess layer.

vlad961 committed:
Deleted redundant layer rules check.
LayerDependencyRulesTest.java includes all necessary tests.

vlad961 committed:
Adjusted ArchitectureTest.java.
Deleted LayerDependencyRulesTest.java.
Refactored L2Violation and moved into service layer.
Refactored TaskService and L9Violation to resemble to other Violations.

vlad961 committed:
Adjusted layer rules.
They equal those of add-layer-rules branch now.

hohwille committed:
Merge branch 'master' into add-violations-of-layer-rules

hohwille committed:
avoid pointless change/diff

hohwille commented:
@vlad961 thanks for this demo PR. Very nice demonstration with good naming of `«RuleId»Violation«Source»LayerDependsOn«Target»Layer` resulting in readable violation output:
```
because Dependency of technical layers violates architecture rules.' was violated (12 times):
Field <com.devonfw.sample.archunit.batch.L11ViolationBatchLayerDependsOnDataaccessLayer.taskItemRepository> has type <com.devonfw.sample.archunit.task.dataaccess.TaskItemRepository> in (L11ViolationBatchLayerDependsOnDataaccessLayer.java:0)
Field <com.devonfw.sample.archunit.batch.L7ViolationBatchLayerDependsOnServiceLayer.taskService> has type <com.devonfw.sample.archunit.task.service.TaskService> in (L7ViolationBatchLayerDependsOnServiceLayer.java:0)
Field <com.devonfw.sample.archunit.client.L3ViolationClientLayerDependsOnLogicLayer.taskItemMapper> has type <com.devonfw.sample.archunit.task.logic.TaskItemMapper> in (L3ViolationClientLayerDependsOnLogicLayer.java:0)
Field <com.devonfw.sample.archunit.client.L4ViolationClientLayerDependsOnDataaccessLayer.taskItemRepository> has type <com.devonfw.sample.archunit.task.dataaccess.TaskItemRepository> in (L4ViolationClientLayerDependsOnDataaccessLayer.java:0)
Field <com.devonfw.sample.archunit.client.L5ViolationClientLayerDependsOnBatchLayer.taskImport> has type <com.devonfw.sample.archunit.batch.SampleTaskImport> in (L5ViolationClientLayerDependsOnBatchLayer.java:0)
Field <com.devonfw.sample.archunit.task.common.L1ViolationCommonLayerDependsOnOtherLayer.taskItemMapper> has type <com.devonfw.sample.archunit.task.logic.TaskItemMapper> in (L1ViolationCommonLayerDependsOnOtherLayer.java:0)
Field <com.devonfw.sample.archunit.task.dataaccess.L10ViolationDataaccessLayerDependsOnServiceLayer.taskService> has type <com.devonfw.sample.archunit.task.service.TaskService> in (L10ViolationDataaccessLayerDependsOnServiceLayer.java:0)
Field <com.devonfw.sample.archunit.task.dataaccess.L12ViolationDataaccessLayerDependsOnLogicLayer.taskItemMapper> has type <com.devonfw.sample.archunit.task.logic.TaskItemMapper> in (L12ViolationDataaccessLayerDependsOnLogicLayer.java:0)
Field <com.devonfw.sample.archunit.task.logic.L9ViolationLogicLayerDependsOnServiceLayer.taskService> has type <com.devonfw.sample.archunit.task.service.TaskService> in (L9ViolationLogicLayerDependsOnServiceLayer.java:0)
Field <com.devonfw.sample.archunit.task.service.L2ViolationNoLayerOtherThanClientDependsOnClientLayer.sampleClient> has type <com.devonfw.sample.archunit.client.SampleClient> in (L2ViolationNoLayerOtherThanClientDependsOnClientLayer.java:0)
Field <com.devonfw.sample.archunit.task.service.L6ViolationServiceLayerDependsOnBatchLayer.sampleTaskImport> has type <com.devonfw.sample.archunit.batch.SampleTaskImport> in (L6ViolationServiceLayerDependsOnBatchLayer.java:0)
Field <com.devonfw.sample.archunit.task.service.L8ViolationServiceLayerDependsOnDataaccessLayer.taskItemRepository> has type <com.devonfw.sample.archunit.task.dataaccess.TaskItemRepository> in (L8ViolationServiceLayerDependsOnDataaccessLayer.java:0)
```

vlad961 committed:
Deleted the unnecessary dump and import script.
Replaced the import script with a simple class declaration,
to demonstrate dependencies towards the batch layer.

hohwille committed:
Merge branch 'master' into add-violations-of-layer-rules

vlad961 committed:
Merge branch 'master' into add-violations-of-layer-rules

hohwille committed:
Merge branch 'violations/layer' into add-violations-of-layer-rules

hohwille committed:
Update L2ViolationNoLayerOtherThanClientDependsOnClientLayer.java
